### PR TITLE
Implemented authenticated GH requests

### PR DESCRIPTION
**Pull request by MatousJobanek**

### CODE REVIEW

File: drone-webdriver/src/test/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSourceAuthenticatedRequestTestCase.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        GeckoDriverGitHubSource geckoDriverGitHubSource = new GeckoDriverGitHubSource(httpClientSpy, gitHubLastUpdateCache, capabilities);
+
+        // when
+        geckoDriverGitHubSource.getReleaseForVersion("v0.14.0");
+
+        // then
+        verify(httpClientSpy, times(1)).get(anyString(), argThat(headers -> {
+            if (headers.size() > 0) {
+                String authorization = headers.get("Authorization");
```
Comment (bartoszmajsak) [inline drone-webdriver/src/test/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSourceAuthenticatedRequestTestCase.java]: Maybe for better readability extract to a new method so we know what is the topic of verification here?

File: drone-webdriver/src/test/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSourceAuthenticatedRequestTestCase.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class GitHubSourceAuthenticatedRequestTestCase {
+
+    private static final Logger log = Logger.getLogger(GitHubSourceAuthenticatedRequestTestCase.class.getName());
+
+    @ClassRule
+    public static HoverflyRule hoverflyRule =
+        HoverflyRule.inSimulationMode(classpath("hoverfly/gh.simulation.mozilla@geckodriver.releases.json"));
```
Comment (bartoszmajsak) [inline drone-webdriver/src/test/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSourceAuthenticatedRequestTestCase.java]: We have `hoverfly-java:0.6.1`, but the latest is `0.8.1` - shall we try to update in this PR?

Comment (MatousJobanek) [inline drone-webdriver/src/test/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSourceAuthenticatedRequestTestCase.java]: I like this updates separated: https://github.com/arquillian/arquillian-extension-drone/pull/109

Comment (MatousJobanek) [inline drone-webdriver/src/test/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSourceAuthenticatedRequestTestCase.java]: But good catch! Thanks for pointing this out. 👍

**Pull request merged**